File: engine/engine-model/src/main/scala/tools/packet/maker/MakerResultDestroy.scala
package tools.packet.maker

import net.opcodes.SendOpcode
import tools.Pair
import tools.packet.PacketInput

class MakerResultDestroy(private var _itemId: Int, private var _mesos: Int,
                         private var _itemsGained: java.util.List[Pair[java.lang.Integer, java.lang.Integer]]) extends PacketInput {
  def itemId: Int = _itemId

  def mesos: Int = _mesos

  def itemsGained: java.util.List[Pair[java.lang.Integer, java.lang.Integer]] = _itemsGained

  override def opcode(): SendOpcode = SendOpcode.MAKER_RESULT
}

File: engine/engine-model/src/main/java/tools/packet/maker/MakerResultDestroy.java
package tools.packet.maker;

import java.util.List;

import net.opcodes.SendOpcode;
import tools.Pair;
import tools.packet.PacketInput;

public record MakerResultDestroy(Integer itemId, Integer mesos,
                                 List<Pair<Integer, Integer>> itemsGained) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MAKER_RESULT;
   }
}